conjure/download.py

```diff
@@ -0,0 +1,68 @@
+import os
+from subprocess import run, CalledProcessError
+
+
+class DownloadException(Exception):
+    pass
+
+
+def remote_exists(path):
+    """ Verifies remote url archive exists
+    """
+    try:
+        run('wget --spider -q {}'.format(path), shell=True,
+            check=True)
+    except CalledProcessError:
+        return False
+    return True
+
+
+def download(src, dst):
+    """ Download and extract archive
+
+    Arguments:
+    src: path to archive
+    dst: directory to change to before extract, this directory must already
+         exist.
+    """
+    try:
+        run("wget -qO- {} | bsdtar -xf - -s'|[^/]*/||' -C {}".format(
+            src, dst), shell=True, check=True)
+    except CalledProcessError as e:
+        raise DownloadException("Unable to download {}: {}".format(src, e))
+
+
+def cycle_locations(path):
```

Comment:
This name was confusing when I read it where it was being called.
I see what it does here, but I think it'd read better if it was called like 'get_remote_url' or something that makes it clearer what it will return…